src/functional/swing_slide_filter.zig

```diff
@@ -70,20 +70,24 @@ pub fn compressSwing(
     var current_timestamp: usize = 2;
     while (current_timestamp < uncompressed_values.len) : (current_timestamp += 1) {
         // Evaluate the upper and lower bound linear functions at current timestamp.
-        const upper_limit = evaluateLinearFunctionAtTime(upper_bound_linear_function, current_timestamp);
-        const lower_limit = evaluateLinearFunctionAtTime(lower_bound_linear_function, current_timestamp);
+        const upper_limit = evaluateLinearFunctionAtTime(
+            upper_bound_linear_function,
+            @floatFromInt(current_timestamp),
```

Comment:
Can the use of `f64` for indices cause problems such a `f64` being unable to represent an `usize` value since there theoretically can be up to `usize` values to compress and `@bitSizeOf(f64) == @bitSizeOf(usize)` on some platforms, e.g., x86-64. Also if a problem occurs will it be detected and an error returned or will incorrectly compressed data be returned? If incorrectly compressed will be returned, additional checks should be added to prevent this even if it adds a significant amount of code complexity and reduce runtime performance as a compression method should never return corrupted data without making it clear that an error has occurred. If a significant amount of code complexity and runtime checks are required to make continues time safe to use in Swing, maybe revert back to using discrete time for Swing and add `ContinousPoint` for Slide.

Reply:
Indeed, I think I will follow this comment https://github.com/cmcuza/TerseTS/pull/38#discussion_r1633848505, close this PR, and create another where Swing does not change its implementation. Moreover, since Point will be a function, I will need to make the Segments a function. This deserves its PR.